app/Legacy/Legacy.php

```diff
@@ -109,27 +109,30 @@ public static function isLegacyModelID(string $id): bool
 	 * @throws ConfigurationException thrown, if the translation between
 	 *                                legacy and modern IDs is disabled
 	 */
-	private static function translateLegacyID(int $id, string $tableName, Request $request): ?string
+	private static function translateLegacyID(int $id, string $tableName, Request $request): string|null
 	{
 		try {
-			$newID = DB::table($tableName)
+			/** @var string $newID */
+			$newID = (string) DB::table($tableName)
 				->where('legacy_id', '=', intval($id))
```

Comment:
```suggestion
			$newID = (string) DB::table($tableName)
				->where('legacy_id', '=', $id)
```

Type annotation and type cast are not necessary as type of `$newID` can be inferred from code and `$id` is already an integer.

Moreover, I don't like the type cast `(string)` because it adds more complexity to the following code than necessary. But I know, that this is required by PhpStan and PHP does not yet support casts to nullable type. I hope https://wiki.php.net/rfc/nullable-casting will make it into PHP 8.x soon.

Reply:
Solved myself in https://github.com/LycheeOrg/Lychee/pull/1336/commits/d4355783d73378e6d9e92717864416702980a7af.